File: Data Structures & Algorithms/two-integer-sum/submission-1.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> prevMap = new HashMap<>();

        for(int x = 0; x < nums.length; x++){
            int num = nums[x];
            int diff = target - num;

            if(prevMap.containsKey(diff)){
                return new int []{prevMap.get(diff), x};
            }
            prevMap.put(num, x);
        }
        return new int [] {};
    }
}

File: Data Structures & Algorithms/jump-game/submission-1.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        memo = {}

        def dp(i):
            if i == len(nums)-1:
                return True
            if nums[i] == 0:
                return False
            if i in memo:
                return memo[i]
            for j in range(1, nums[i]+1):
                if dp(i+j):
                    return True
            return False
        return dp(0)